include/iocore/net/ConnectionTracker.h

```diff
@@ -166,9 +175,9 @@ class ConnectionTracker
     void blocked();
     /// Clear all reservations.
     void clear();
-    /// Drop the reservation - assume it will be cleaned up elsewhere.
+    /// Transfer ownership of the group outside of this state.
     /// @return The group for this reservation.
-    Group *drop();
+    std::shared_ptr<Group> drop();
```

Comment:
Why return the group? The original code needed that to track the memory because it was an intrusive container and the pointer was useful for putting it back. If it's reference counted, Otherwise it's useless computation.

Reply:
So the calling code can share ownership of the group. That's the point of the call. Once the reference count goes to zero, the group is removed from the table.